core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -25,8 +25,19 @@
 
   private static class LibcLoader {
 
+    /* These values come from Graal's imageinfo API which aims to offer the ability to detect
+     * when we're in the Graal build/run time via system props.  The maintainers of Graal have
+     * agreed that this API will not change over time.  We reference these props as literals
+     * to avoid introducing a dependency on Graal code for non-Graal users here. */
+    private static final String GRAAL_STATUS_PROP = "org.graalvm.nativeimage.imagecode";
+    private static final String GRAAL_RUNTIME_STATUS = "runtime";
+
     public Libc load() {
```

Comment:
I was considering adding a config option that would allow the user to disable this functionality even for Graal VM apps if they desired.  A user might want to do something like this if they don't want to install the LLVM toolchain for some reason (I'm not sure why that would be an issue but let's assume it is) or they don't want to rely on a newer feature.

Is this worthwhile?  I can see pretty clear arguments for and against... and the more I think about it the more it seems like it might not be worth it.

Reply:
Yeah I agree the use case sounds contrived... Let's keep it like this for now, if someone needs it they will ask for it.